build/helpers/coverage.js

```diff
@@ -24,19 +24,18 @@ const session = new Session();
 
 const metrics = [
   {
-    ...Arithmetic,
-    weight: 1 / 3
+    ...Arithmetic
   },
   {
-    ...Byte,
-    weight: 1 / 3
+    ...Byte
   },
   {
-    ...Logical,
-    weight: 1 / 3
+    ...Logical
   }
 ];
 
+let weight = metrics.length;
+
```

Comment:
There's quite a bit of global state introduced in this PR. It's nothing that a bit of refactoring can't fix though; be smart about passing state around as function parameters and you should be able to get rid of the global state.